src/pybind/mgr/dashboard/frontend/src/app/shared/services/api-interceptor.service.ts

```diff
@@ -66,10 +66,22 @@ export class ApiInterceptorService implements HttpInterceptor {
 
           let timeoutId;
           if (showNotification) {
+            // Get the error message. If the error is triggered by the backend,
+            // then the 'resp.error' is an object containing the 'detail' key.
+            // Otherwise 'resp.error' can be a string. If none of them appears,
+            // check if 'message' exists and is a string.
+            const message =
```

Comment:
Maybe it's just me but this nested conditions aren't easy to read. Could you please refactor them to be more readable?

Reply:
Any suggestion how you would like to see this? Thanks to prettify which does the indention i think it is clear enough what happens here. Using if else will not increase readability in my opinion. Also there is a documentation above the code which should enlight the code reader, too.

Reply:
My suggestion would be to do it like so:

```typescript
if (_.isPlainObject(resp.error) && _.isString(resp.error.detail)) {
	message = resp.error.detail;
} else if (_.isString(resp.error)) {
	message = resp.error;
} else if (_.isString(resp.message)) {
	message = resp.message;
} else {
	message = '';
}
```

This is (likely) the way most people are familiar with writing and reading conditions. Though admittedly, I'm likely biased towards writing conditions that way, although it's more text to parse. But you wouldn't have added a comment to "my" version of the condition, would you? That's a good indicator for what's clear or not. If you feel like you have to add a comment, it doesn't seem to be clear. If you can rewrite it in way where you don't think that you'll need an extra comment describing what's being done, I'd tend to say it's clearer.

Nonetheless, I just got used to how you did it, although my opinion didn't change.

Reply:
Done